experiments/tootsie/exp1395_qwen3_32b.py

```diff
@@ -0,0 +1,67 @@
+"""
+Switch over to Qwen3 architecture, warmstarting from the llama-32b-tootsie
+"""
+
+import dataclasses
+
+import haliax
+from levanter.optim import AdamConfig
+from levanter.optim.clip_update_norm import ClipUpdateNormConfig
+
+from experiments.defaults import default_train
+from experiments.qwen3 import qwen3_32b
+from experiments.tootsie.exp1295_32b import llama_32b_tootsie, llama_32b_train_config, nemotron_mix
+from marin.execution import executor_main
+from marin.resources import TpuPodConfig
+
+# We have doctored the opt state to include update history from
+# gs://marin-us-central2/checkpoints/llama-32b-tootsie-2/checkpoints/step-77096 for clipping
+warmstart_checkpoint = llama_32b_tootsie.cd("checkpoints/step-80000/").nonblocking()
+
+qwen3_32b_remat = dataclasses.replace(
+    qwen3_32b, gradient_checkpointing=haliax.ScanCheckpointPolicy(save_carries="offload")
+)
+
+
+qwen_32b_warmstart_train = dataclasses.replace(
+    llama_32b_train_config,
+    initialize_from_checkpoint_path=warmstart_checkpoint,
+    resources=TpuPodConfig("v4-2048", 1),
+    reset_data_loader_on_init=False,
+    allow_partial_checkpoint=True,
+    optimizer_config=AdamConfig(
+        beta1=0.9,
+        beta2=0.95,
+        epsilon=1e-8,
+        max_grad_norm=0.2,  # we're almost always < .2 except during spikes
```

Comment:
This seems likely to change during cooldown based on: https://arxiv.org/abs/2506.02285

but we can cross that bridge when we come to it.

Reply:
good point! though if we do the AdamC thing it should be fine right?